client/src/scene.rs

```diff
@@ -1,7 +1,190 @@
-use crate::instance;
-use crate::model;
+use crate::camera::Camera;
+use crate::{instance, resources};
+use crate::instance::Instance;
+use crate::model::{self, InstancedModel, Model};
+use std::collections::HashMap;
+use std::cell::Cell;
 
-pub struct Scene {
+use nalgebra_glm as glm;
+
+pub enum ModelIndices{
+    ISLAND = 0,
+    PLAYER = 1,
+    CUBE = 2,
+    FERRIS = 3,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ModelIndex{
+    pub index: usize,
+}
+
+#[derive(Clone, Debug)]
+pub struct Node {
+    // pub parent: Box<Node>,
+    pub childnodes: Vec<Node>,
```

Comment:
it's kind of minor but can we merge the parallel vectors into one just to simplify the code a bit? The type could be something like `Vec<(Node, Transform)>`.

Reply:
Yea great idea! I can do the same for the models and modeltransforms.